affinity/cpp-20/d0796r1.md

```diff
@@ -112,20 +117,21 @@ There are some additional challenges which we have been investigating but are no
 
     /* Execution resource */
 
-    struct execution_resource {
+    class execution_resource {
+     public:
 
       execution_resource() = delete;
-      execution_resource(const execution_resource &) = delete;
-      execution_resource(execution_resource &&) = delete;
-      execution_resource &operator=(const execution_resource &) = delete;
-      execution_resource &operator=(execution_resource &&) = delete;
+      execution_resource(const execution_resource &);
+      execution_resource(execution_resource &&);
+      execution_resource &operator=(const execution_resource &);
+      execution_resource &operator=(execution_resource &&);
       ~execution_resource();
 
       size_t concurrency() const noexcept;
-      size_t partition_size() const noexcept;
 
-      const execution_resource &partition(size_t i) const noexcept;
-      const execution_resource &member_of() const noexcept;
+      std::vector<resource> resources() const noexcept;
```

Comment:
define an alias for this in a single place. We'll likely have to change this later anw

Reply:
I feel creating an alias for this will make the wording more complex as we'd have to specify what kind of interface the type provides. We have a note that we may want to change this type in the future.